src/components/shop-app.js

```diff
@@ -346,32 +344,33 @@ class ShopApp extends connect(store)(LitElement) {
   }
 
   firstUpdated() {
-    installRouter((location) => this._updateLocation(location));
+    installRouter((location) => store.dispatch(updateLocation(location)));
     installOfflineWatcher((offline) => store.dispatch(updateNetworkStatus(offline)));
     installMediaQueryWatcher('(max-width: 767px)', (matches) => this._smallScreen = matches);
 
     // Custom elements polyfill safe way to indicate an element has been upgraded.
     this.removeAttribute('unresolved');
   }
 
-  _stateChanged(state) {
-    const category = currentCategorySelector(state);
+  stateChanged(state) {
     this._page = state.app.page;
-    this._categories = state.categories;
+    this._categories = Object.keys(state.categories).map(key => state.categories[key]);
```

Comment:
Any reason to move `Object.keys(...)`here? This gets executed any time the store updates opposite of only when the properties that shop-app cares get update.